### PR TITLE
[nikobus] open connection on refresh even if no modules exist

### PR DESCRIPTION
**Pull request by gitwouter**
Signed-off-by: Wouter Denayer <wouter@denayer.com>

open connection on refresh even if no modules exist

currently, when no modules are defined the connection to nikobus is not opened and therefore the button presses are not detected.

### CODE REVIEW

File: bundles/org.openhab.binding.nikobus/src/main/java/org/openhab/binding/nikobus/internal/handler/NikobusPcLinkHandler.java [hunk 1]
```diff
@@ -329,6 +329,22 @@ private void refresh() {
...
+        // if there are command listeners then we need an open connection even if no modules exist
+        if (!commandListeners.isEmpty()) {
+            NikobusConnection connection = this.connection;
+            if (connection == null) {
+                return;
+            }        	
+            try {
+                connectIfNeeded(connection);
+            } catch (IOException e) {
+                logger.debug("refresh failed due {}", e.getMessage(), e);
```
Comment (fwolter) [inline bundles/org.openhab.binding.nikobus/src/main/java/org/openhab/binding/nikobus/internal/handler/NikobusPcLinkHandler.java:342]: The log message could be removed, as the status update is already logged by the framework.

The state change originated by updateStatus() is logged to events.log. Including the status detail message.

**Review by fwolter — CHANGES_REQUESTED** [review]

File: bundles/org.openhab.binding.daikin/deploy.sh [hunk 2]
```diff
@@ -0,0 +1,2 @@
+cp ./target/org.openhab.binding.daikin-3.1.0-SNAPSHOT.jar /Volumes/openhab/share/addons/
```
Comment (fwolter) [inline bundles/org.openhab.binding.daikin/deploy.sh:1]: Can you remove this?

**Review by fwolter — CHANGES_REQUESTED** [review]

File: bundles/org.openhab.binding.daikin/deploy.sh (hunk 2 above)
Comment (gitwouter) [reply bundles/org.openhab.binding.daikin/deploy.sh:1]: oops, yes

**Review by crnjan — APPROVED** [review]
lgtm

File: bundles/org.openhab.binding.nikobus/src/main/java/org/openhab/binding/nikobus/internal/handler/NikobusPcLinkHandler.java [hunk 3]
```diff
@@ -329,6 +329,20 @@ private void refresh() {
...
+            NikobusConnection connection = this.connection;
+            if (connection == null) {
+                return;
+            }
+            try {
+                connectIfNeeded(connection);
+            } catch (IOException e) {
+                connection.close();
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, e.getMessage());
+            }
```
Comment (cpmeister) [inline bundles/org.openhab.binding.nikobus/src/main/java/org/openhab/binding/nikobus/internal/handler/NikobusPcLinkHandler.java:343]: Should this function return at this point? Or should the handler still be refreshed even if the thing is offline?

Comment (gitwouter) [reply bundles/org.openhab.binding.nikobus/src/main/java/org/openhab/binding/nikobus/internal/handler/NikobusPcLinkHandler.java]: good point, continuing does not make much sense if the connection is not available.

**Comment by fwolter** [comment]
You need to sign-off your commits. See https://www.openhab.org/docs/developer/contributing.html#sign-your-work The necessary commands are listed when you click on "Details" at the DCO check below.

**Review by cpmeister — APPROVED** [review]

**Review by fwolter — APPROVED** [review]

**Pull request merged**